client_test.go

```diff
@@ -173,3 +178,92 @@ func TestRequest_200UnmarshalPut(t *testing.T) {
 		t.Errorf("expected: name; got: %s", result.Name)
 	}
 }
+
+func TestClient_requestWithRetries(t *testing.T) {
+	// It looks like if the call to c.client.Do doesn't read the body
+	// before returning an error, then bodyBuffer never gets populated
+	// and every retry ends up with a blank body.
```

Comment:
This caption makes no sense without historical context
```suggestion
	// Test that calls to c.client.Do will retry correctly,
	// even if the original request fails prematurely
```

Reply:
Good point! Yeah, I wrote it down before the fix and then never got back to update it.